### ISSUE TITLE
Extract permutation from dataset after shuffling

### ISSUE THREAD

**Issue by benjamin9555** [rusty1s/pytorch_geometric#292]
## 🚀 Feature

After calling `dataset.shuffle`, it would be good if you could extract the [torch random permutation](https://rusty1s.github.io/pytorch_geometric/build/html/_modules/torch_geometric/data/in_memory_dataset.html#InMemoryDataset.shuffle)  for reproducibility.

## Motivation

If I want to exactly recreate an experiment or continue training a model, I need to able to provide the same train/val/test split of the data.

**Comment by rusty1s** [comment]
Good idea! Fixed in master (see [here](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/data/in_memory_dataset.py#L97-L99))
